rpc/jsonrpc/client/http_json_client.go

```diff
@@ -21,13 +21,16 @@ const (
 	protoWSS   = "wss"
 	protoWS    = "ws"
 	protoTCP   = "tcp"
+	protoUNIX  = "unix"
```

Comment:
Wasn't aware we supported unix here. 🤯

Reply:
me neither, then I saw a test fail from here https://github.com/tendermint/tendermint/blob/master/rpc/jsonrpc/jsonrpc_test.go#L33

and I realized why we were hacking the paths with `.`  :(

Reply:
We should probably look at dropping this as we should keep the surface area of this part of the codebase small. This can be part of the RPC refactor.

Reply:
if the plan is still to support an http rest API running via unix sockets, I'd suggest we have a look at how docker does it, to get a clearer idea on how they plug together the socket path, the request path and all the URL parsing flow

Reply:
Can we make a note of these ideas in a follow up issue so it doesn't get lost when we merge this PR?